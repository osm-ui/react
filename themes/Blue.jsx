
import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import _ from 'lodash';
import colors from '../colors';
import { buildDarkThemeConfig } from '../helpers/themes';
import {
    config as defaultThemeConfig,
    StyledDiv,
} from './Default';


export const config = _.merge(
    {},
    buildDarkThemeConfig(defaultThemeConfig, colors, 'blue'),
    {},
);


const RedTheme = ({ children, ...rest }) => (
    <ThemeProvider theme={config}>
        <StyledDiv {...rest}>
            {children}
        </StyledDiv>
    </ThemeProvider>
);


RedTheme.propTypes = {
    children: PropTypes.node.isRequired,
};

RedTheme.displayName = 'RedTheme';

export default RedTheme;
