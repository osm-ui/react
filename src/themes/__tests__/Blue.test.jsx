import React from 'react';
import {
  snapshotWithElementChildren,
  snapshotWithTextChildren
} from 'helpers/tests';
import Blue from 'themes/Blue';

describe('When using snapshots', () => {
  it('Should render with an element children', () =>
    snapshotWithElementChildren(Blue));

  it('Should render with a text children', () =>
    snapshotWithTextChildren(Blue));
});
