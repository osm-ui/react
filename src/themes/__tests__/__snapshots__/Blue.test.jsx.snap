// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using snapshots Should render with a text children 1`] = `
<Default__StyledSpan>
  <BlueTheme
    key=".0"
  >
    This is a text
  </BlueTheme>
</Default__StyledSpan>
`;

exports[`When using snapshots Should render with an element children 1`] = `
<Default__StyledSpan>
  <BlueTheme
    key=".0"
  >
    <div />
  </BlueTheme>
</Default__StyledSpan>
`;
